### PR TITLE
Publish workflow on CI

### PR DESCRIPTION
**Pull request by Jrigada**
# Publish workflow on CI

## Description

Publish workflow

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: publish
+
+on:
+  push:
+    tags:        
+      - '*'           # Push events to every tag not containing /
+      # Allow manual triggering
+  workflow_dispatch:
+  pull_request:
+    branches: [ '*' ]
```
Comment (Oppen) [inline .github/workflows/publish.yml:10]: Remove this one please. We don't want a deploy per PR. We only want to check semver there.

Comment (Jrigada) [reply .github/workflows/publish.yml:10]: Of course. It was just for debugging purposes

File: Cargo.toml [hunk 2]
```diff
@@ -3,15 +3,15 @@ members = [".", "felt", "./deps/parse-hyperlinks"]
...
 edition = "2021"
-license = "MIT"
+license = "Apache-2.0"
 description = "Blazing fast Cairo interpreter"
 
 # See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
 
 [features]
-default = ["with_mimalloc"]
+default = ["with_mimalloc"]
```
Comment (Oppen) [inline Cargo.toml:14]: Remove the trailing whitespace.

File: Cargo.toml [hunk 3]
```diff
@@ -44,7 +44,8 @@ keccak = "0.1.2"
 # There is a proposal for extending nom::delimited to use this function:
 # https://github.com/Geal/nom/issues/1253
 parse-hyperlinks = { path = "./deps/parse-hyperlinks", version = "0.23.4" }
-felt = { package = "cairo-felt", path = "./felt", version = "0.1.0" }
+felt = { package = "cairo-felt", version = "0.1.1" }
```
Comment (Oppen) [inline Cargo.toml:47]: Keep the path for development. Otherwise we'll need to publish for every change or have ad-hoc changes.

**Review by Oppen — CHANGES_REQUESTED** [review]

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      run: cargo publish --token ${CRATES_TOKEN} --all-features --manifest-path ./felt/Cargo.toml
+    # FIXME: there should be a better way to make sure the index in crates.io is updated before publishing
+    # cairo-vm but right now the step throws timeout and fails. 
+    - name: wait for index in crates.io
+      run: sleep 120
+    - name: Publish crate cairo-vm
+      env:
+          CRATES_TOKEN: ${{ secrets.CARGO_REGISTRY_TOKEN }}
+      run: cargo publish --token ${CRATES_TOKEN} --all-features -p cairo-vm
+
```
Comment (fmoletta) [inline .github/workflows/publish.yml:33]: Add newline at EOF

**Review by Oppen — APPROVED** [review]

**Pull request merged**